### PR TITLE
adding request body logic in global exception handler

### PR DESCRIPTION
**Pull request by RAJKUMAR733**

### CODE REVIEW

File: src/main/java/com/banking/poc/migration/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -41,8 +42,9 @@ public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
     }
     @ExceptionHandler(MigratedAccountNotFoundException.class)
     ResponseEntity<String>migratedAccountNotFoundException(MigratedAccountNotFoundException
-                                                             migratedAccountNotFoundException){
-        return new ResponseEntity<String>("Record is not found!",HttpStatus.NOT_FOUND);
+                                                             migratedAccountNotFoundException, HttpServletRequest httpServletRequest){
+        String accountId=httpServletRequest.getAttribute("accountId").toString();
+        return new ResponseEntity<String>("Migrated account with id "+accountId+" is not found!",HttpStatus.NOT_FOUND);
```
Comment (samratshekhar) [inline src/main/java/com/banking/poc/migration/exception/GlobalExceptionHandler.java:47]: Tests?

File: src/main/java/com/banking/poc/migration/controller/MigratedAccountController.java [hunk 2]
```diff
@@ -33,8 +34,9 @@ ResponseEntity<MigratedAccountResponseDto> addMigratedAccount(
     @GetMapping
     @TrackExecutionTime
     ResponseEntity<MigratedAccountResponseDto> getMigratedAccount(
-            @Valid @RequestBody MigratedAccountRequestDto migratedAccountRequestDto)
+            @Valid @RequestBody MigratedAccountRequestDto migratedAccountRequestDto, HttpServletRequest httpServletRequest)
             throws MigratedAccountNotFoundException {
+        httpServletRequest.setAttribute("accountId", migratedAccountRequestDto.getAccountId());
```
Comment (samratshekhar) [inline src/main/java/com/banking/poc/migration/controller/MigratedAccountController.java:39]: To check if we can create an AOP method to set standard attributes.

File: src/test/java/com/banking/poc/migration/controller/MigratedAccountControllerTest.java [hunk 3]
```diff
@@ -104,14 +133,76 @@ void testGetMigratedElementNotFound() throws MigratedAccountNotFoundException {
...
                 assertEquals(HttpStatus.OK, response.getStatusCode());
         }
 
-    @Test
-    void testMigratedAccountNotPresentException() throws MigratedAccountNotFoundException {
-        when(migratedAccountService.getMigratedAccount("1")).thenThrow(new MigratedAccountNotFoundException("Migrated Not Found"));
-        assertThrows(MigratedAccountNotFoundException.class,()->migratedAccountService.getMigratedAccount("1"));
-    }
+        @Test
+        void testMigratedAccountNotPresentException() throws MigratedAccountNotFoundException {
```
Comment (samratshekhar) [inline src/test/java/com/banking/poc/migration/controller/MigratedAccountControllerTest.java:141]: What is this test supposed to do?

Comment (RAJKUMAR733) [reply src/test/java/com/banking/poc/migration/controller/MigratedAccountControllerTest.java:141]: It is checking, is it throwing MigratedAccountException or not

**Review by samratshekhar — APPROVED** [review]

**Pull request merged**